tests/conftest.py

```diff
@@ -283,17 +283,35 @@ def init_kwargs(request):
 
 
 @pytest.fixture(scope="function")
-def task(request, init_kwargs):
+def task(request, generate_task):
     """Gets a task instance."""
-    # set default values used for the initialization of the task.
-    init_args = {
-        "new_task_name": "",
-    }
+    new_task_name = _get_value_from_pytest_markers("new_task_name", request)
+    new_task_name = "" if new_task_name is None else new_task_name
+    return generate_task(task_name=new_task_name)
 
-    # iterate through markers and update arguments
+
+@pytest.fixture(scope="function")
+def generate_task(init_kwargs):
+    """Gets a factory function which can be used to generate new tasks."""
```

Comment:
I don't see the distinction between generating tasks and creating tasks, and would prefer to call this `create_task`.

Reply:
Unless the distinction is that this automatically cleans up the task?

Please add a warning to the fixture docstring like https://github.com/ni/nidaqmx-python/blob/master/tests/unit/conftest.py#L26

Reply:
I have added proper documentation to denote that the cleanup is also handled by the same fixture.